### PR TITLE
Reuse ADB instance

### PR DESCRIPTION
**Pull request by roman-mazur**
When we force a new debug bridge, `kill-server` command will be sent to an existing instance.
There are some difficulties to properly start adb server after this.
This results in https://github.com/stanfy/spoon-gradle-plugin/issues/4

Also `waitForAdb` implementation is changed to check whether devices list is ready.
See
https://android.googlesource.com/platform/tools/base/+/gradle_1.1.3/build-system/builder/src/main/java/com/android/builder/testing/ConnectedDeviceProvider.java#60

### CODE REVIEW

**Comment by roman-mazur** [comment]
With this change I was finally able to run spoon tests for different falvors in one gradle build.
Fixes stanfy/spoon-gradle-plugin#4.

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonUtils.java [hunk 1]
```diff
@@ -9,6 +9,9 @@
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
 import com.madgag.gif.fmsware.AnimatedGifEncoder;
+import org.apache.commons.io.FileUtils;
+
+import javax.imageio.ImageIO;
```
Comment (roman-mazur) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonUtils.java]: Sorry for this... But it seems to be correct, does not it?

Comment (edenman) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonUtils.java]: Please revert.  We keep regular imports in a simple alphabetical list, with no newlines between them.

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonUtils.java [hunk 2]
```diff
@@ -129,17 +130,19 @@ static void createAnimatedGif(List<File> testScreenshots, File animatedGif) thro
   }
 
   private static void waitForAdb(AndroidDebugBridge adb) {
-    for (int i = 1; i < 10; i++) {
+    long timeOut = 30000; // 30 sec
```
Comment (edenman) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonUtils.java]: kill the comment and replace with `TimeUnit.SECONDS.toMillis(30)`.  Same for the next line.  And it's not a hard requirement but I'd rename them to `timeOutMs` and `sleepTimeMs`.

**Comment by edenman** [comment]
LG, just a couple nits.

**Comment by roman-mazur** [comment]
Updated, PTAL.

**Comment by roman-mazur** [comment]
@edenman Do you think you'll afford another release?

**Comment by edenman** [comment]
@roman-mazur just released 1.1.7.  trade you for a spoon-gradle-plugin release? :)

**Comment by roman-mazur** [comment]
Yes, just want to close one issue before 1.0 release.

**Pull request merged**